lib/system/extra.ex

```diff
@@ -0,0 +1,84 @@
+defmodule System.Extra do
+  @moduledoc """
+  Extensions to the standard library's System module
+
+  """
+
+  alias __MODULE__.CmdStreamer
+
+
+
+  #########################################################
+  # Types
+  #########################################################
+
+  @typep acc :: any
+  @typep element :: any
+
+
+
+  #########################################################
+  # Public API
+  #########################################################
+
+  @doc """
+  Asserts that the provided path is valid and an executable binary.
+
+  iex> System.Extra.assert_executable!("/usr/local/bin/mix")
```

Comment:
This seems really fragile. Could we at least do /bin/sh or something slightly more common?

Reply:
Please at least use `$PATH`. on my machine:
```
❯ which mix
/usr/bin/mix
```

Reply:
Great feedback, @jdemaris & @glittershark !